extensions/avro-extensions/src/main/java/io/druid/data/input/avro/GenericRecordAsMap.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.druid.data.input.avro;
+
+import org.apache.avro.generic.GenericRecord;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+public class GenericRecordAsMap implements Map<String, Object>
```

Comment:
Can you add a comment here as to why there are so many unsupported operation exceptions? as in, they were simply not needed for the initial use case, but may be added later? or simply can never be added?